src/ReinforcementLearningCore/src/core/hooks.jl

```diff
@@ -104,28 +104,28 @@ Base.getindex(h::RewardsPerEpisode) = h.rewards
 Store the total reward of each episode in the field of `rewards`. If
 `is_display_on_exit` is set to `true`, a unicode plot will be shown at the [`PostExperimentStage`](@ref).
 """
-mutable struct TotalRewardPerEpisode{T} <: AbstractHook where {T<:Union{Bool,Nothing}}
+struct TotalRewardPerEpisode{T} <: AbstractHook where {T<:Union{Bool,Nothing}}
     rewards::Vector{Float64}
-    reward::Float64
+    reward::Vector{Float64}
```

Comment:
I suppose the goal here is to let this be mutated. It is a better idea to use `reward::Ref{Float64}` then, because a vector allocates memory for an array, even when empty. Plus, it is a bit confusing to the reader why this is a one-element vector. It works with empty indexing:
![image](https://user-images.githubusercontent.com/47037088/224052758-ab4bb234-b96c-458e-b1e5-a2ea25f3034a.png)

Reply:
Nice! I didn't know about this, thanks!

Reply:
After looking into this a bit more, the most reasonable approach is leaving it the way it was, mutable keyword yields the desired mutability and doesn't cause performance issues.